cli/azd/pkg/containerapps/container_app.go

```diff
@@ -124,6 +130,28 @@ func (cas *containerAppService) DeployYaml(
 		return fmt.Errorf("decoding yaml: %w", err)
 	}
 
+	if shouldPersist := cas.alphaFeatureManager.IsEnabled(persistCustomDomainsFeature); shouldPersist {
+		aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
```

Comment:
I'm nervous about the general "read and then update our model" plan.  It feels like it is walking us down a path we don't want, where the source of truth is not what was in the App Host.

If it is critical that we solve this before we can move project generation into the AppHost the way we think we'd want to, I'd much rather we do so by using the metadata feature we have in the manifest and figure out a way to expose this on `AddProject()` in line with what we might think things would look like long term.

I am worried this is pushing us down the wrong path where the source of truth is not in code, it's in the foreign resource and changes made in the portal are sometimes reflected here, I also think it's going to make things harder for us to get where we eventually want to be with the AppHost generating this IaC.

Reply:
> where the source of truth is not what was in the App Host

I think that's the main idea and justification behind the strategy on this PR.  The idea:

Within the universe of what you can do, either from your AppHost or directly with bicep, there is not a 100% IaC way to model custom domains, unless you are using Azure DNS Zones to manage your domains.

So, this PR is modeling a world where ACA custom domains are disconnected from the IaC and considered a manual-only configuration. Neither model or affected by IaC.

Reply:
I think we can do IaC but we should try to be precise. The flow is:
- Deploy the app
- Get information from deployed app about what to configure in your DNS provider
- Add a certificate this requires verifying that you own the domain if you are using managed certificate. This adds a certificate as a sub resource to the container app environment
- That certificate id needs to be referenced by the custom domains section in the container app

Reply:
@davidfowl , on the second bullet, where it says:

> configure in your DNS provider

I think that's the fundamental issue why IaC can't describe the full scenario.

Reply:
Can IaC do the steps after I do the manual step?

Reply:
> Manual step

But we can model this manual step in IaC.  For example, if the manual step is "you need to go take this string from an output of the CAE object and go add it to the DNS for your custom-domain so we can validate that you own it", you can model this in ARM using a deploymentScript (perhaps using the `forceUpdateTag` to ensure it re-runs if the output from the CAE ever changes).  The ACA could `dependsOn` this resource to make sure the script runs before we try to create the ACA?

Now, it may be that you just literally can't write that deployment script yourself (perhaps your DNS provider doesn't have an API, and you have to go to some cloud portal to make changes) and we'd have to figure out what we want to do there, but it feels like there should be some programmatic way to stitch this together.  We just need a way to model "there's a special step here"

Reply:
TBH I think we are only just at the beginning of our journey being able to produce the Bicep for the container apps inside Aspire. Without this feature its quite painful to apply a custom domain to an Aspire/AZD deployed app.

Your choices are:

1. `azd infra synth` and hack away at the Bicep (commiting yourself to hand merging changes).
2. Deploy a reverse proxy in front of the service (AFD doesn't support websockets which is a problem for blazor) and then there is the cost factor to consider of that as well.

The other aspect of this is that this whole DNS record thing is very Azure specific. If I was deploying my app to a K8S cluster I could just point the DNS later on outside of k8s.